shared/hooks/useSearch.ts

```diff
@@ -20,16 +22,20 @@ const useSearch = (allPossibleResults: SearchResult[], query: string, mode?: 'as
         prefix: true,
       },
     })
-    if (mode === 'async') {
-      search.addAllAsync(allPossibleResults)
+    if (asyncMode) {
+      search.addAllAsync(allPossibleResults).then(() => setIsIndexing(false))
     } else {
       search.addAll(allPossibleResults)
     }
     return search
-  }, [allPossibleResults, mode])
+  }, [allPossibleResults, asyncMode])
 
-  // @ts-expect-error minisearch doesn't add the returned storeFields (e.g. title or path) to its typing
-  return query.length === 0 ? allPossibleResults : minisearch.search(query)
+  useEffect(() => {
+    // @ts-expect-error minisearch doesn't add the returned storeFields (e.g. title or path) to its typing
+    setSearchResult(query.length === 0 ? allPossibleResults : miniSearch.search(query))
+  }, [allPossibleResults, isIndexing, miniSearch, query])
```

Comment:
This looks a little bit complicated to me. Can't we just use `useLoadAsync` instead of `useMemo`, `useEffect` and two `useState`s?
```ts
const useSearch = (allPossibleResults: SearchResult[], query: string): SearchResult[] => {
  const initializeMiniSearch = useCallback(async () => {
    const search = new MiniSearch({
      idField: 'path',
      fields: ['title', 'content'],
      storeFields: ['title', 'content', 'path', 'thumbnail'],
      searchOptions: {
        boost: { title: 2 },
        fuzzy: true,
        prefix: true,
      },
    })
    await search.addAllAsync(allPossibleResults)
    return search
  }, [allPossibleResults])

  const { data: minisearch } = useLoadAsync(initializeMiniSearch)

  return query.length === 0 || !minisearch ? allPossibleResults : minisearch.search(query)
}
```